spikeinterface/extractors/nwbextractors.py

```diff
@@ -95,15 +107,43 @@ class NwbRecordingExtractor(BaseRecording):
     installation_mesg = "To use the Nwb extractors, install pynwb: \n\n pip install pynwb\n\n"
     name = "nwb"
 
-    def __init__(self, file_path: PathType, electrical_series_name: str = None, load_time_vector: bool = False,
-                 samples_for_rate_estimation: int = 100000, driver=None):
-
-        self.driver = driver
+    def __init__(
+        self, 
+        file_path: PathType, 
+        electrical_series_name: str = None, 
+        load_time_vector: bool = False,
+        samples_for_rate_estimation: int = 100000, 
+        stream_mode: str = None, 
+        stream_cache_path: PathType = None
+    ):
         check_nwb_install()
-        self._file_path = str(file_path)
+        self.stream_mode = stream_mode
+        self.stream_cache_path = stream_cache_path
```

Comment:
should validate inputs. stream_cache_path should only be defined if stream_mode == "fsspec"